### ISSUE TITLE
Roslyn build errors VS2017

### ISSUE THREAD

**Issue by doorman02** [dotnet/roslyn#29992]
**Version Used**: 
VS2017

**Steps to Reproduce**:

I am getting Roslyn diagnostics errors when parsing a very basic .NET 4.6 application. The solution is a xaml application with two dependent projects. The dependency tree looks like this:

UI -> BLL -> DB

I am getting the following diagnostics errors in the BLL project:
```
Predefined type ´System.Object´ is not defined or imported
Predefined type ´System.Void´ is not defined or imported
Predefined type ´System.Int32´ is not defined or imported
'object´does not contain a constructor that takes 0 arguments.
```

The solution and projects build fine, still Roslyn gives these errors. Maybe the errors are misleading and I need to configure the projects in someway? Any idea how I can resolve these errors?

**Expected Behavior**:
Roslyn should also be able to build solution.

**Actual Behavior**:
Roslyn build results in errors

**Comment by jaredpar** [comment]
Those errors indicate that the compiler is not being passed the correct set of references. Generally this happens when a project has not been properly restored before building. Did you validate that the restore succeeded?

**Comment by doorman02** [comment]
Hi @jaredpar yes you are correct subscriping to the workspace.workspaceFailed event results in erros. Please see the updated issue above.

**Comment by jaredpar** [comment]
I do not think this is a bug but instead an issue in how the Workspace is being constructed. I can't comment much more than that though. Will have to wait for @jinujoseph

**Comment by doorman02** [comment]
Hi @jaredpar I actually solved it tonight. I think there was a problem with the wrong msbuild version being used during runtime of my unittests. https://stackoverflow.com/questions/52385005/strange-diagnostics-errors-prefedined-type-system-is-not-defined-or-imported

**Comment by DustinCampbell** [comment]
@doorman02 : If you've moved over to using the `Microsoft.Build.Locator` as suggested in that Stack Overflow post, then yes, that is the correct solution.
